test/sagemaker_tests/pytorch/__init__.py

```diff
@@ -0,0 +1,169 @@
+    target_ecr_client = boto3.client("ecr", region_name=target_region)
+    target_account_id = sts_client.get_caller_identity().get("Account")
+    image_account_id = get_account_id_from_image_uri(source_image_uri)
+    image_region = get_region_from_image_uri(source_image_uri)
+    image_repo_uri, image_tag = source_image_uri.split(":")
+    _, image_repo_name = image_repo_uri.split("/")
+    if not ecr_repo_exists(target_ecr_client, target_image_repo_name):
+        raise ECRRepoDoesNotExist(
+            f"Repo named {target_image_repo_name} does not exist in {target_region} on the account {target_account_id}"
+        )
+
+    target_image_uri = (
+        source_image_uri.replace(image_region, target_region)
+        .replace(image_repo_name, target_image_repo_name)
+        .replace(image_account_id, target_account_id)
+    )
+
+    client = boto3.client('ecr', region_name = image_region)
+    username, password = get_ecr_login_boto3(client, image_account_id, image_region)
+    save_credentials_to_file(ECR_PASSWORD_FILE_PATH, password)
+
+    # using ctx.run throws error on codebuild "OSError: reading from stdin while output is captured".
+    # Also it throws more errors related to awscli if in_stream=False flag is added to ctx.run which needs more deep dive
+    subprocess.check_output(f"cat {ECR_PASSWORD_FILE_PATH} | docker login -u {username} --password-stdin https://{image_account_id}.dkr.ecr.{image_region}.amazonaws.com && docker pull {source_image_uri}", shell=True, executable="/bin/bash")
+    subprocess.check_output(f"docker tag {source_image_uri} {target_image_uri}", shell=True, executable="/bin/bash")
+    delete_file(ECR_PASSWORD_FILE_PATH)
+    username, password = get_ecr_login_boto3(target_ecr_client, target_account_id, target_region)
+    save_credentials_to_file(ECR_PASSWORD_FILE_PATH, password)
+    subprocess.check_output(f"cat {ECR_PASSWORD_FILE_PATH} | docker login -u {username} --password-stdin https://{target_account_id}.dkr.ecr.{target_region}.amazonaws.com && docker push {target_image_uri}", shell=True, executable="/bin/bash")
+    delete_file(ECR_PASSWORD_FILE_PATH)
+
+    return target_image_uri
+
+def get_ecr_image(ecr_image, region):
+    """
+    It uploads image to n_virginia region and return image uri
+    """
+    image_repo_uri, image_tag = ecr_image.split(":")
+    _, image_repo_name = image_repo_uri.split("/")
+    target_image_repo_name = f"{image_repo_name}"
+    regional_ecr_image = reupload_image_to_test_ecr(ecr_image, target_image_repo_name, region)
+    return regional_ecr_image
+
+
+def invoke_pytorch_helper_function(pdx_ecr_image, sagemaker_region, helper_function, helper_function_args):
+
+    RETRY = 2
+    DELAY = 300
+    for _ in range(RETRY):
+        for region in sagemaker_region:
+            sagemaker_session = get_sagemaker_session(region)
+            ecr_image = get_ecr_image(pdx_ecr_image, region) if region != "us-west-2" else pdx_ecr_image
```

Comment:
can we make more dynamic for any region passed?
Where is pdx_ecr_image defined? i can't find it in PR changes by search